### PR TITLE
Minor refactors to `init_dist_connection`

### PR DESCRIPTION
**Pull request by ananthsub**
From going through the code in prep for https://github.com/PyTorchLightning/pytorch-lightning/issues/11725
- We can fail faster by raising the runtime error first
- Remove a level of nesting and return earlier if torch distributed is already initialized

### Does your PR introduce any breaking changes? If yes, please list them.
No

## Before submitting

- [n/a] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @awaelchli @rohitgr7 @akihironitta @justusschock

### CODE REVIEW

File: pytorch_lightning/utilities/distributed.py [hunk 1]
```diff
@@ -366,26 +366,30 @@ def init_dist_connection(
...
         kwargs: kwargs for ``init_process_group``
+
+    Raises:
+        RuntimeError:
+            If ``torch.distributed`` is not available
     """
+    if not torch.distributed.is_available():
+        raise RuntimeError("torch.distributed is not available. Cannot initialize distributed process group")
+    if torch.distributed.is_initialized():
+        log.debug("torch distributed is already initialized. exiting early")
```
Comment (ananthsub) [inline pytorch_lightning/utilities/distributed.py:377]:
```suggestion
        log.debug("torch.distributed is already initialized. Exiting early")
```

File: pytorch_lightning/utilities/distributed.py [hunk 2]
```diff
@@ -366,26 +366,30 @@ def init_dist_connection(
...
-        )
-
-        # on rank=0 let everyone know training is starting
-        rank_zero_info(
-            f"{'-' * 100}\n"
-            f"distributed_backend={torch_distributed_backend}\n"
-            f"All distributed processes registered. Starting with {world_size} processes\n"
-            f"{'-' * 100}\n"
-        )
+    log.info(f"initializing distributed: GLOBAL_RANK: {global_rank}, MEMBER: {global_rank + 1}/{world_size}")
```
Comment (ananthsub) [inline pytorch_lightning/utilities/distributed.py:383]:
```suggestion
    log.info(f"Initializing distributed: GLOBAL_RANK: {global_rank}, MEMBER: {global_rank + 1}/{world_size}")
```

**Review by kaushikb11 — APPROVED** [review]
LGTM!

**Review by rohitgr7 — APPROVED** [review]

**Review by carmocca — APPROVED** [review]
